core/src/worker/mod.rs

```diff
@@ -121,8 +126,206 @@ pub struct Worker {
     local_activities_complete: Arc<AtomicBool>,
     /// Used to track all permits have been released
     all_permits_tracker: tokio::sync::Mutex<AllPermitsTracker>,
-    /// Used to shutdown the worker heartbeat task
-    worker_heartbeat: Option<WorkerHeartbeatManager>,
+    worker_heartbeat_data: Option<Arc<Mutex<WorkerHeartbeatData>>>,
```

Comment:
Similar issue here - `WorkerHeartbeatData` is mixing a lot of concerns. Leave data as just data, and separate out the behavior. My comments on `capture_heartbeat` / `fetch_config` get at that.

Like I had in my comment on the last review round - we can just have `get_worker_heartbeat_data()`, which you do have, but we don't need to actually store data - you can just construct a new one every time it's called.